File: osm/addr-osmify-java/src/test/java/hu/vmiklos/addr_osmify/MockUrlopen.java
/*
 * Copyright 2019 Miklos Vajna
 *
 * SPDX-License-Identifier: MIT
 */

package hu.vmiklos.addr_osmify;

/**
 * Sets and restores App.urlopener for testing.
 */
public class MockUrlopen
{
    Urlopener old;
    MockUrlopen(String suffix)
    {
        old = App.urlopener;
        App.urlopener = new MockUrlopener(suffix);
    }

    void destruct() { App.urlopener = old; }
}

/* vim:set shiftwidth=4 softtabstop=4 expandtab: */

File: osm/addr-osmify-kotlin/src/test/kotlin/MockUrlopen.kt
/*
 * Copyright 2020 Miklos Vajna
 *
 * SPDX-License-Identifier: MIT
 */

package hu.vmiklos.addr_osmify

import hu.vmiklos.addr_osmify.App.Companion.urlopener

/**
 * Sets and restores App.urlopener for testing.
 */
class MockUrlopen internal constructor(suffix: String) {
    var old: Urlopener
    fun destruct() {
        urlopener = old
    }

    init {
        old = urlopener
        urlopener = MockUrlopener(suffix)
    }
}

/* vim:set shiftwidth=4 softtabstop=4 expandtab: */
